From linux-usb-devel-bounces@lists.sourceforge.net Thu Dec 14 10:50:18 2006
Date: Thu, 14 Dec 2006 19:49:35 +0100
From: Miguel Angel Alvarez  <ma.alvarez@ziv.es>
To: linux-usb-devel@lists.sourceforge.net
Message-ID: <20061214184935.GE16801@kiste.smurf.noris.de>
Subject: USB: fix interaction between different interfaces in an "Option" usb device

From: Miguel Angel Alvarez  <ma.alvarez@ziv.es>

Just the serial port in the first interface should control DTR and RTS
lines. This way, the closing of the rest of the ports does not produce a=
hangup in the communication.

Signed-off-by: Miguel Angel Alvarez <ma.alvarez@ziv.es>
Signed-off-by: Matthias Urlichs <matthias@urlichs.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/option.c |    3 +++
 1 file changed, 3 insertions(+)

--- gregkh-2.6.orig/drivers/usb/serial/option.c
+++ gregkh-2.6/drivers/usb/serial/option.c
@@ -625,6 +625,9 @@ static int option_send_setup(struct usb_
 
 	dbg("%s", __FUNCTION__);
 
+	if (port->number != 0)
+		return 0;
+
 	portdata = usb_get_serial_port_data(port);
 
 	if (port->tty) {
